packages/app-builder-lib/src/targets/LinuxTargetHelper.ts

```diff
@@ -61,13 +61,11 @@ export class LinuxTargetHelper {
       sources.push(icnsPath)
     }
 
-    // if no explicit sources are defined, default to buildResources directory
-    if (sources.length < 1) {
-      sources.push('./')
```

Comment:
Actually, I was the one who made this change. The reason why I didn't put `packager.buildResourcesDir` in `fallbackSources` is because in this case, it's not actually fallback (we just assume the dev wants to use the globally defined `buildResourcesDir`). I think fallback means `electron-builder` find no defined icons and opt to use Electron icon instead (Ref: https://github.com/electron-userland/electron-builder/blob/master/packages/app-builder-lib/src/platformPackager.ts#L590).

Reply:
Hmmm, that log warning seems to be assuming that only one fallback source is provided.

It's possible to provide multiple fallback sources, would that be sequentially read? To provide the functionality: 1st. icns -> 2nd. build resources -> 3rd. framework icon

With the code as-is, the `snapTests` need fixing. :/

Reply:
I just checked more carefully, it seems like as long as `fallbackSources` is used, `isFallback = true`. Ref: https://github.com/develar/app-builder/blob/1cceb8fd6fe68955cecb96fb48259087980cd826/pkg/icons/icon-converter.go#L64

Reply:
> With the code as-is, the `snapTests` need fixing. :/

You mean https://github.com/electron-userland/electron-builder/pull/5385 breaks `snapTests`?

Reply:
In this case, I would propose to keep `buildResourcesDir` in `sources` and make a small change:
`sources.push('./')` -> `sources.push(packager.buildResourcesDir)`

Reply:
Yes #5385 broke the test suite. Seems `linuxPackagerTest` and/or `snapTest` looking at my terminal history.

I'm confused why we can't use fallback sources? 
The `isFallback` (from what you've linked) is simply used in the code for logging purposes, after which the value is discarded.  Functionally, nothing behaves differently. We can update the logged message and/or make the boolean flag more explicit as to what source was fallen back to

I just tried `sources.push(packager.buildResourcesDir)` and it doesn't work due to other build errors as everything expects a icon `sources` path to always exist, whereas fallback sources seem to behave as optional.
```
 icon directory /private/var/folders/xv/37hyp31j7tv32txvjnnly0p1qccy3t/T/et-a34345c0e457c6fd5301936e13688492/t-O76XSJ/test-project-4/build doesn't contain icons
```

Reply:
+1 If so, then feel free to keep your changes. I still feel it's not what `fallbackSources` is intended to be. But since it works perfectly and you have worked so hard on all those changes, it's unnecessary to make things harder.